src/launch/main.c

```diff
@@ -585,7 +599,7 @@ static int manager_on_set_activation_environment(Manager *manager, sd_bus_messag
         if (r < 0)
                 return error_origin(r);
 
-        r = sd_bus_send(manager->bus_regular, method_call, NULL);
+        r = sd_bus_call_async(manager->bus_regular, NULL, method_call, manager_set_environment_handler, NULL, -1);
```

Comment:
Urgs, what are the lifetime rules for floating slots again? They're destroyed when we drop our last bus-reference? If that's true, then this looks ok. I know, we don't pass a userdata context, but still...

Reply:
I don't see the issue either way. There is nothing for the slot to protect... But yes, lifetime is the same as the bus, or the timeout (which we set to -1).